File: cassandrablog/src/main/webapp/app/entities/cassandrablog/cass-saathratri-entity-3/service/cass-saathratri-entity-3.service.ts
import { HttpClient, HttpParams, HttpResponse, httpResource } from '@angular/common/http';
import { Injectable, computed, inject, signal } from '@angular/core';

import dayjs from 'dayjs/esm';
import { Observable, map } from 'rxjs';

import { ApplicationConfigService } from 'app/core/config/application-config.service';
import { createRequestOption } from 'app/core/request/request-util';
import { isPresent } from 'app/core/util/operators';
import { ICassSaathratriEntity3, ICassSaathratriEntity3Id, NewCassSaathratriEntity3 } from '../cass-saathratri-entity-3.model';
export type PartialUpdateCassSaathratriEntity3 = Partial<ICassSaathratriEntity3> & Pick<ICassSaathratriEntity3, 'compositeId'>;

type RestOf<T extends ICassSaathratriEntity3 | NewCassSaathratriEntity3> = Omit<T, 'createdTimeId' | 'departureDate' | 'tags'> & {
  compositeId: {};
  departureDate?: number | null;
  tags?: Set<string> | null;
};

export type RestCassSaathratriEntity3 = RestOf<ICassSaathratriEntity3>;

export type NewRestCassSaathratriEntity3 = RestOf<NewCassSaathratriEntity3>;

export type PartialUpdateRestCassSaathratriEntity3 = RestOf<PartialUpdateCassSaathratriEntity3>;

export type EntityResponseType = HttpResponse<ICassSaathratriEntity3>;
export type EntityArrayResponseType = HttpResponse<ICassSaathratriEntity3[]>;

@Injectable()
export class CassSaathratriEntity3sService {
  readonly cassSaathratriEntity3sParams = signal<
    Record<string, string | number | boolean | readonly (string | number | boolean)[]> | undefined
  >(undefined);
  readonly cassSaathratriEntity3sResource = httpResource<RestCassSaathratriEntity3[]>(() => {
    const params = this.cassSaathratriEntity3sParams();
    if (!params) {
      return undefined;
    }
    return { url: this.resourceUrl, params };
  });
  /**
   * This signal holds the list of cassSaathratriEntity3 that have been fetched.
   */
  readonly cassSaathratriEntity3s = computed(() =>
    (this.cassSaathratriEntity3sResource.hasValue() ? this.cassSaathratriEntity3sResource.value() : []).map(item =>
      this.convertValueFromServer(item),
    ),
  );
  protected readonly applicationConfigService = inject(ApplicationConfigService);
  protected readonly resourceUrl = this.applicationConfigService.getEndpointFor('api/cass-saathratri-entity-3-s', 'cassandrablog');

  protected convertValueFromServer(restCassSaathratriEntity3: RestCassSaathratriEntity3): ICassSaathratriEntity3 {
    return {
      ...restCassSaathratriEntity3,
      compositeId: {
        entityType: restCassSaathratriEntity3.compositeId.entityType,

        createdTimeId: restCassSaathratriEntity3.compositeId.createdTimeId,
      },
      departureDate: restCassSaathratriEntity3.departureDate ? dayjs(restCassSaathratriEntity3.departureDate) : null,
      tags: restCassSaathratriEntity3.tags ? new Set(restCassSaathratriEntity3.tags) : new Set<string>(),
    };
  }
}

@Injectable({ providedIn: 'root' })
export class CassSaathratriEntity3Service extends CassSaathratriEntity3sService {
  protected readonly http = inject(HttpClient);

  create(cassSaathratriEntity3: NewCassSaathratriEntity3): Observable<EntityResponseType> {
    const copy = this.convertValueFromClient(cassSaathratriEntity3);
    return this.http
      .post<RestCassSaathratriEntity3>(this.resourceUrl, copy, { observe: 'response' })
      .pipe(map(res => this.convertResponseFromServer(res)));
  }

  update(cassSaathratriEntity3: ICassSaathratriEntity3): Observable<EntityResponseType> {
    const copy = this.convertValueFromClient(cassSaathratriEntity3);
    return this.http
      .put<RestCassSaathratriEntity3>(
        `${this.resourceUrl}/${cassSaathratriEntity3.compositeId.entityType}/${cassSaathratriEntity3.compositeId.createdTimeId}`,
        copy,
        { observe: 'response' },
      )
      .pipe(map(res => this.convertResponseFromServer(res)));
  }

  partialUpdate(cassSaathratriEntity3: PartialUpdateCassSaathratriEntity3): Observable<EntityResponseType> {
    const copy = this.convertValueFromClient(cassSaathratriEntity3);
    return this.http
      .patch<RestCassSaathratriEntity3>(
        `${this.resourceUrl}/${cassSaathratriEntity3.compositeId.entityType}/${cassSaathratriEntity3.compositeId.createdTimeId}`,
        copy,
        { observe: 'response' },
      )
      .pipe(map(res => this.convertResponseFromServer(res)));
  }

  find(entityType: string, createdTimeId: string): Observable<EntityResponseType> {
    const data = { entityType, createdTimeId };
    return this.http
      .get<RestCassSaathratriEntity3>(`${this.resourceUrl}/get`, { params: data, observe: 'response' })
      .pipe(map(res => this.convertResponseFromServer(res)));
  }

  query(req?: any): Observable<EntityArrayResponseType> {
    const options = createRequestOption(req);
    return this.http
      .get<RestCassSaathratriEntity3[]>(this.resourceUrl, { params: options, observe: 'response' })
      .pipe(map(res => this.convertResponseArrayFromServer(res)));
  }

  querySlice(req?: any): Observable<EntityArrayResponseType> {
    const options = createRequestOption(req);
    return this.http
      .get<RestCassSaathratriEntity3[]>(`${this.resourceUrl}/slice`, { params: options, observe: 'response' })
      .pipe(map(res => this.convertResponseArrayFromServer(res)));
  }

  findAllByCompositeIdEntityTypePageable(entityType: string, req?: any): Observable<EntityArrayResponseType> {
    let options = createRequestOption(req);
    options = options.set('entityType', entityType);
    return this.http
      .get<
        RestCassSaathratriEntity3[]
      >(`${this.resourceUrl}/find-all-by-composite-id-entity-type-pageable`, { params: options, observe: 'response' })
      .pipe(map(res => this.convertResponseArrayFromServer(res)));
  }

  findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdPageable(
    entityType: string,
    createdTimeId: string,
    req?: any,
  ): Observable<EntityArrayResponseType> {
    let options = createRequestOption(req);
    options = options.set('entityType', entityType);
    options = options.set('createdTimeId', createdTimeId);
    return this.http
      .get<
        RestCassSaathratriEntity3[]
      >(`${this.resourceUrl}/find-all-by-composite-id-entity-type-and-composite-id-created-time-id-pageable`, { params: options, observe: 'response' })
      .pipe(map(res => this.convertResponseArrayFromServer(res)));
  }

  findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdLessThanPageable(
    entityType: string,
    createdTimeId: string,
    req?: any,
  ): Observable<EntityArrayResponseType> {
    let options = createRequestOption(req);
    options = options.set('entityType', entityType);
    options = options.set('createdTimeId', createdTimeId);
    return this.http
      .get<
        RestCassSaathratriEntity3[]
      >(`${this.resourceUrl}/find-all-by-composite-id-entity-type-and-composite-id-created-time-id-less-than-pageable`, { params: options, observe: 'response' })
      .pipe(map(res => this.convertResponseArrayFromServer(res)));
  }

  findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdLessThanEqualPageable(
    entityType: string,
    createdTimeId: string,
    req?: any,
  ): Observable<EntityArrayResponseType> {
    let options = createRequestOption(req);
    options = options.set('entityType', entityType);
    options = options.set('createdTimeId', createdTimeId);
    return this.http
      .get<
        RestCassSaathratriEntity3[]
      >(`${this.resourceUrl}/find-all-by-composite-id-entity-type-and-composite-id-created-time-id-less-than-equal-pageable`, { params: options, observe: 'response' })
      .pipe(map(res => this.convertResponseArrayFromServer(res)));
  }

  findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdGreaterThanPageable(
    entityType: string,
    createdTimeId: string,
    req?: any,
  ): Observable<EntityArrayResponseType> {
    let options = createRequestOption(req);
    options = options.set('entityType', entityType);
    options = options.set('createdTimeId', createdTimeId);
    return this.http
      .get<
        RestCassSaathratriEntity3[]
      >(`${this.resourceUrl}/find-all-by-composite-id-entity-type-and-composite-id-created-time-id-greater-than-pageable`, { params: options, observe: 'response' })
      .pipe(map(res => this.convertResponseArrayFromServer(res)));
  }

  findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdGreaterThanEqualPageable(
    entityType: string,
    createdTimeId: string,
    req?: any,
  ): Observable<EntityArrayResponseType> {
    let options = createRequestOption(req);
    options = options.set('entityType', entityType);
    options = options.set('createdTimeId', createdTimeId);
    return this.http
      .get<
        RestCassSaathratriEntity3[]
      >(`${this.resourceUrl}/find-all-by-composite-id-entity-type-and-composite-id-created-time-id-greater-than-equal-pageable`, { params: options, observe: 'response' })
      .pipe(map(res => this.convertResponseArrayFromServer(res)));
  }

  findByCompositeIdEntityTypeAndCompositeIdCreatedTimeId(entityType: string, createdTimeId: string): Observable<EntityResponseType> {
    let options = new HttpParams();
    options = options.set('entityType', entityType);
    options = options.set('createdTimeId', createdTimeId);
    return this.http
      .get<RestCassSaathratriEntity3>(`${this.resourceUrl}/find-by-composite-id-entity-type-and-composite-id-created-time-id`, {
        params: options,
        observe: 'response',
      })
      .pipe(map(res => this.convertResponseFromServer(res)));
  }

  delete(cassSaathratriEntity3: ICassSaathratriEntity3): Observable<HttpResponse<{}>> {
    const copy = this.convertValueFromClient(cassSaathratriEntity3);
    return this.http.delete(
      `${this.resourceUrl}/${cassSaathratriEntity3.compositeId.entityType}/${cassSaathratriEntity3.compositeId.createdTimeId}`,
      { observe: 'response' },
    );
  }

  getCassSaathratriEntity3Identifier(cassSaathratriEntity3: Pick<ICassSaathratriEntity3, 'compositeId'>): ICassSaathratriEntity3Id {
    return cassSaathratriEntity3.compositeId;
  }
  compareCassSaathratriEntity3(
    o1: Pick<ICassSaathratriEntity3, 'compositeId'> | null,
    o2: Pick<ICassSaathratriEntity3, 'compositeId'> | null,
  ): boolean {
    return o1 && o2 ? this.getCassSaathratriEntity3Identifier(o1) === this.getCassSaathratriEntity3Identifier(o2) : o1 === o2;
  }
  addCassSaathratriEntity3ToCollectionIfMissing<Type extends Pick<ICassSaathratriEntity3, 'compositeId'>>(
    cassSaathratriEntity3Collection: Type[],
    ...cassSaathratriEntity3sToCheck: (Type | null | undefined)[]
  ): Type[] {
    const cassSaathratriEntity3s: Type[] = cassSaathratriEntity3sToCheck.filter(isPresent);
    if (cassSaathratriEntity3s.length > 0) {
      const cassSaathratriEntity3CollectionIdentifiers = cassSaathratriEntity3Collection.map(cassSaathratriEntity3Item =>
        this.getCassSaathratriEntity3Identifier(cassSaathratriEntity3Item),
      );
      const cassSaathratriEntity3sToAdd = cassSaathratriEntity3s.filter(cassSaathratriEntity3Item => {
        const cassSaathratriEntity3Identifier = this.getCassSaathratriEntity3Identifier(cassSaathratriEntity3Item);
        if (cassSaathratriEntity3CollectionIdentifiers.includes(cassSaathratriEntity3Identifier)) {
          return false;
        }
        cassSaathratriEntity3CollectionIdentifiers.push(cassSaathratriEntity3Identifier);
        return true;
      });
      return [...cassSaathratriEntity3sToAdd, ...cassSaathratriEntity3Collection];
    }
    return cassSaathratriEntity3Collection;
  }

  protected convertValueFromClient<T extends ICassSaathratriEntity3 | NewCassSaathratriEntity3 | PartialUpdateCassSaathratriEntity3>(
    cassSaathratriEntity3: T,
  ): RestOf<T> {
    return {
      ...cassSaathratriEntity3,
      compositeId: {
        ...cassSaathratriEntity3.compositeId,
      },
      departureDate: cassSaathratriEntity3.departureDate ? cassSaathratriEntity3.departureDate.valueOf() : null,
      tags: cassSaathratriEntity3.tags ? Array.from(cassSaathratriEntity3.tags) : [],
    } as RestOf<T>;
  }

  protected convertResponseFromServer(res: HttpResponse<RestCassSaathratriEntity3>): HttpResponse<ICassSaathratriEntity3> {
    return res.clone({
      body: res.body ? this.convertValueFromServer(res.body) : null,
    });
  }

  protected convertResponseArrayFromServer(res: HttpResponse<RestCassSaathratriEntity3[]>): HttpResponse<ICassSaathratriEntity3[]> {
    return res.clone({
      body: res.body ? res.body.map(item => this.convertValueFromServer(item)) : null,
    });
  }
}

File: cassandrablog/src/main/java/com/saathratri/developer/cassandra/blog/service/CassSaathratriEntity3Service.java
package com.saathratri.developer.cassandra.blog.service;

import com.saathratri.developer.cassandra.blog.domain.CassSaathratriEntity3Id;
import com.saathratri.developer.cassandra.blog.service.dto.CassSaathratriEntity3DTO;
import java.util.List;
import java.util.Optional;
import java.util.UUID;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;

/**
 * Service Interface for managing {@link com.saathratri.developer.cassandra.blog.domain.CassSaathratriEntity3}.
 */
public interface CassSaathratriEntity3Service {
    /**
     * Save a cassSaathratriEntity3.
     *
     * @param cassSaathratriEntity3DTO the entity to save.
     * @return the persisted entity.
     */
    CassSaathratriEntity3DTO save(CassSaathratriEntity3DTO cassSaathratriEntity3DTO);

    /**
     * Updates a cassSaathratriEntity3.
     *
     * @param cassSaathratriEntity3DTO the entity to update.
     * @return the persisted entity.
     */
    CassSaathratriEntity3DTO update(CassSaathratriEntity3DTO cassSaathratriEntity3DTO);

    /**
     * Partially updates a cassSaathratriEntity3.
     *
     * @param cassSaathratriEntity3DTO the entity to update partially.
     * @return the persisted entity.
     */
    Optional<CassSaathratriEntity3DTO> partialUpdate(CassSaathratriEntity3DTO cassSaathratriEntity3DTO);

    /**
     * Get all the cassSaathratriEntity3s.
     *
     * @return the list of entities.
     */
    List<CassSaathratriEntity3DTO> findAll();

    /**
     * Get the "id" cassSaathratriEntity3.
     *
     * @param id the id of the entity.
     * @return the entity.
     */
    Optional<CassSaathratriEntity3DTO> findOne(CassSaathratriEntity3Id id);

    /**
     * Delete the "id" cassSaathratriEntity3.
     *
     * @param id the id of the entity.
     */
    void delete(CassSaathratriEntity3Id id);

    /**
     * Get all the cassSaathratriEntity3s with Cassandra cursor-based pagination.
     *
     * @param pageable the pagination information.
     * @return the slice of entities.
     */
    Slice<CassSaathratriEntity3DTO> findAllSlice(org.springframework.data.domain.Pageable pageable);

    List<CassSaathratriEntity3DTO> findAllByCompositeIdEntityType(final String entityType);
    Slice<CassSaathratriEntity3DTO> findAllByCompositeIdEntityTypePageable(final String entityType, Pageable pageable);
    Optional<CassSaathratriEntity3DTO> findByCompositeIdEntityTypeAndCompositeIdCreatedTimeId(
        final String entityType,
        final UUID createdTimeId
    );
    List<CassSaathratriEntity3DTO> findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdLessThan(
        final String entityType,
        final UUID createdTimeId
    );
    Slice<CassSaathratriEntity3DTO> findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdLessThanPageable(
        final String entityType,
        final UUID createdTimeId,
        Pageable pageable
    );
    List<CassSaathratriEntity3DTO> findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdLessThanEqual(
        final String entityType,
        final UUID createdTimeId
    );
    Slice<CassSaathratriEntity3DTO> findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdLessThanEqualPageable(
        final String entityType,
        final UUID createdTimeId,
        Pageable pageable
    );
    List<CassSaathratriEntity3DTO> findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdGreaterThan(
        final String entityType,
        final UUID createdTimeId
    );
    Slice<CassSaathratriEntity3DTO> findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdGreaterThanPageable(
        final String entityType,
        final UUID createdTimeId,
        Pageable pageable
    );
    List<CassSaathratriEntity3DTO> findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdGreaterThanEqual(
        final String entityType,
        final UUID createdTimeId
    );
    Slice<CassSaathratriEntity3DTO> findAllByCompositeIdEntityTypeAndCompositeIdCreatedTimeIdGreaterThanEqualPageable(
        final String entityType,
        final UUID createdTimeId,
        Pageable pageable
    );
    CassSaathratriEntity3DTO findLatestByCompositeIdEntityType(final String entityType);
}
